autospec/infile_handler.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/true
+#
+# infile_parser.py - part of autospec
+# Copyright (C) 2017 Intel Corporation
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+import os
+import requests
+import tempfile
+from urllib.request import urlretrieve
+
+import util
+import infile_bb_parser
+
+
+# filetypes and their parse order (lower parsed first)
+parsable_filetypes = {'.inc': 1, '.bb': 2}
+
+
+def parse_ext(path):
+    ext = os.path.splitext(path)[1]
+    if ext not in parsable_filetypes:
+        util.print_warning("Cannot parse infile \"{}\" type. "
+                           "from input: {}".format(ext, path))
+        return
+
+    if ext.startswith('.'):
+        ext = ext[1:]
+
+    return ext
+
+
+def check_url_content(url):
+    if "text/html" in requests.head(url).headers['content-type']:
+        if 'git' in url and "/tree/" in url:
+            url = url.replace("/tree/", "/plain/", 1)
+        else:
+            util.print_fatal("infile url has an html content-type, "
+                             "please use plaintext.")
+            return None
+    return url
+
+
+def sort_files(x):
+    return parsable_filetypes.get(os.path.splitext(x)[1])
+
+
+def parser_type(bb_fp, output_dict, parse_type):
+    if parse_type == "bb":
+        return infile_bb_parser.bb_scraper(bb_fp, output_dict)
+    elif parse_type == "inc":
+        return infile_bb_parser.bb_scraper(bb_fp, output_dict)
+
+
+def file_handler(indata, output_dict):
+    """
+    This function determines whether the input is a file or a url. If it is a
+    url it checks that it is in the correct format (plaintext), downloads the
+    url to a temporary file and passes the file handler to be scraped.
+    If the input is a file, then it opens the file and passes the handler to
+    be scraped.
+
+    The type of parsing bitbake, inc, deb, etc is based on the file extension.
+    """
+
+    parse_type = parse_ext(indata)
+
+    if output_dict.get('filename'):
+        output_dict['filename'].append(indata)
+    else:
+        output_dict['filename'] = [indata]
+
+    if not os.path.isfile(indata):
+        # check that input is plain or raw text and not html
+        indata = check_url_content(indata)
```

Comment:
`indata` might be set to `None` by `check_url_content` if the url is not for a plaintext file. What happens in `urlretrieve` if this happens?

Reply:
Done